reactive/src/main/scala/colibri/reactive/Reactive.scala

```diff
@@ -158,39 +159,43 @@ object Var {
   def combine[A](read: Rx[A], write: RxWriter[A]): Var[A] = new VarCombine(read, write)
 
   @inline implicit class SeqVarOperations[A](rxvar: Var[Seq[A]]) {
-    def sequence(implicit owner: Owner): Rx[Seq[Var[A]]] = Rx.observableSync(new Observable[Seq[Var[A]]] {
-
-      def unsafeSubscribe(sink: Observer[Seq[Var[A]]]): Cancelable = {
-        rxvar.observable.unsafeSubscribe(
-          Observer.create(
-            { seq =>
-              sink.unsafeOnNext(seq.zipWithIndex.map { case (a, idx) =>
-                val observer   = new Observer[A] {
-                  def unsafeOnNext(value: A): Unit = {
-                    rxvar.set(seq.updated(idx, value))
-                  }
 
-                  def unsafeOnError(error: Throwable): Unit = {
-                    sink.unsafeOnError(error)
+    def sequence(implicit owner: Owner): Rx[Seq[Var[A]]] = {
+      val observable = new Observable[Seq[Var[A]]] {
+        def unsafeSubscribe(sink: Observer[Seq[Var[A]]]): Cancelable = {
+          // keep a var for every index of the original sequence
+          val vars = mutable.ArrayBuffer.empty[Var[A]]
+
+          def createIndexVar(idx:Int, seed: A):Var[A] = {
+            Var[A](seed).transformVarRxWriter{ _.contramap{ newValue => 
```

Comment:
We need to be careful here, because we are using the outer owner here! Because this indexed var will never be cancelled. Maybe ok, because it might still be garbage collected, but still. These hot vars with their owners are really difficult to develop with safely.

Is this change urgent? If yes, we can try to fix it. If not, I would prefer to merge #271 first :)

Reply:
Ok, it's not urgent

Reply:
done :)